tests/testthat/test-paren_body_linter.R

```diff
@@ -0,0 +1,46 @@
+testthat::test_that("paren_body_linter returns correct lints", {
+  linter <- paren_body_linter()
+  msg <- "There should be a space between right parenthesis and a body expression."
+
+  # No space after the closing parenthesis prompts a lint
+  expect_lint("function()test", msg, linter)
+  expect_lint("print('hello')\nx <- function(x)NULL\nprint('hello')", msg, linter)
+  expect_lint("if (TRUE)test", msg, linter)
+  expect_lint("while (TRUE)test", msg, linter)
+  expect_lint("for (i in seq_along(1))test", msg, linter)
+  expect_lint("\\()test", msg, linter)
```

Comment:
The lambda syntax should only be checked in R >= 4.0.
Curiously, the test also fails for R 4.0.

Can you investigate?

Reply:
for failure on 4.0 itself, wasn't there a bug with utils::getParseData() on that release?

Reply:
I am going to look into running that particular test only for R>=4.0 and will check if the parsed tree is somehow wrong for version 4.0

Reply:
Regarding the failure on the R version 4.0.5 (I assume that's the one running there). The new lambda syntax was introduced in the version 4.1.0 as per this NEWS document: https://cran.r-project.org/doc/manuals/r-release/NEWS.pdf (3rd point on the 3rd page).

Reply:
My bad re the version number. The condition needs to be on R version >= 4.1.0, not 4.0.0
Do you need help with adding the linter to the defaults?

Reply:
I have added the necessary skip condition.